tests/test_PySurfacePlane_GpuMem.py

```diff
@@ -0,0 +1,326 @@
+#
+# Copyright 2024 Yves33
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Starting from Python 3.8 DLL search policy has changed.
+# We need to add path to CUDA DLLs explicitly.
+import sys
+import os
+from os.path import join, dirname
+
+__DBG_IMG__ = False
+
+if os.name == "nt":
+    # Add CUDA_PATH env variable
+    cuda_path = os.environ["CUDA_PATH"]
+    if cuda_path:
+        os.add_dll_directory(cuda_path)
+    else:
+        print("CUDA_PATH environment variable is not set.", file=sys.stderr)
+        print("Can't set CUDA DLLs search path.", file=sys.stderr)
+        exit(1)
+
+    # Add PATH as well for minor CUDA releases
+    sys_path = os.environ["PATH"]
+    if sys_path:
+        paths = sys_path.split(";")
+        for path in paths:
+            if os.path.isdir(path):
+                os.add_dll_directory(path)
+    else:
+        print("PATH environment variable is not set.", file=sys.stderr)
+        exit(1)
+
+import PyNvCodec as nvc
+import numpy as np
+import unittest
+import json
+import test_common as tc
+try:
+    import pycuda
+    import pycuda.autoinit
+    PYCUDA_AVAIL=True
+except:
+    print("Pycuda is required to run this test")
+    PYCUDA_AVAIL=False
+
+try:
+    sys.path.insert(0,"../extern/dlpack/apps/numpy_dlpack/dlpack")
+    from dlpack import _c_str_dltensor, DLManagedTensor
+    DLPACK_AVAIL=True
+except:
+    DLPACK_AVAIL=False
+
+tolerance=0.0
+
+## rather than copying data to host for testing GpuMem(),
```

Comment:
Hi @Yves33 

Its much easier to do that via `data_ptr`: https://pytorch.org/docs/stable/generated/torch.Tensor.data_ptr.html

Just create torch cuda tensor from `Surface` with single `SurfacePlane` via DLPack, then compare `Surface.PlanePtr.GpuMem` against tensor `data_ptr`.

Reply:
Yes, but it introduces pytorch dependancy for this unittest (whereas dlpack.py is supposed to be already present in extern dependancies).

Are you sure you prefer pytorch way?

Reply:
It’s my personal opinion based on a trade off between test complexity and easiness of integration.

Pytorch is imported in another test anyway.

However, it’s your PR so you decide.

Reply:
Given the size of a full torch install, I think it's better to avoid adding torch dependency.
(And, as you stated, there is already an example of interoperability with torch).